lib/ts/config.ts

```diff
@@ -554,8 +592,13 @@ export async function getQuestions(flags: UserFlags) {
         {
             name: "recipe",
             type: "list",
-            message: "What type of authentication do you want to use?",
-            choices: mapOptionsToChoices(recipeOptions),
+            message: (answers: Answers) => getRecipePromptMessage(answers),
+            choices: async (answers: Answers) =>
+                filterChoices(
+                    mapOptionsToChoices(recipeOptions),
+                    answers,
+                    FILTER_CHOICES_STRATEGY.filterRecipeByUiType
+                ),
             when: (answers: Answers) => {
```

Comment:
all existing code also needs to answers only. Not flag

Reply:
Updated the when conditions of all the steps to use the `answers`